src/connection.js

```diff
@@ -481,7 +496,25 @@ class Connection extends EventEmitter {
         deprecateNonBooleanConfigValue('options.useUTC', config.options.useUTC);
         this.config.options.useUTC = config.options.useUTC;
       }
-      deprecateNullConfigValue('options.useUTC', config.options.useUTC);
+
+      // Whenever authentication property is used to specify an authentication method,
+      // the client will request encryption (the default value of the Encrypt property will be true)
+      // and it will validate the server certificate (regardless of the encryption setting), unless TrustServerCertificate = true
+      if (config.options.authentication != undefined) {
+        // check for valid options
+        if (!(config.options.authentication.toUpperCase() === this.fedAuthInfo.ValidFedAuthEnum.SqlPassword ||
+        config.options.authentication.toUpperCase() === this.fedAuthInfo.ValidFedAuthEnum.ActiveDirectoryPassword)) {
+          throw new Error('An invalid authentication method is specified');
+        }
+        if (this.config.options.tdsVersion < '7_4') {
+          throw new Error(`Azure Active Directory authentication is not supported in the TDS version ${this.config.options.tdsVersion}`);
+        }
+        this.config.options.encrypt = true;
+        this.fedAuthInfo.method = config.options.authentication;
+        if (!config.options.trustServerCertificate) {
```

Comment:
`trustServerCertificate` by default is set to `true` for regular cases [here](https://github.com/tediousjs/tedious/pull/707/files#diff-875da84119066ec244742e95d1209f0cR483), but it is set to `false` for AAD. Can we stick with the existing default behaviour here?